### ISSUE TITLE
[//tests/pallas:export_back_compat_pallas_test_gpu] INTERNAL: RET_CHECK failure (external/xla/xla/service/gpu/ir_emitter_unnested.cc:1388) triton_module Failed to parse Triton module: ML�R

### ISSUE THREAD

**Issue by gpupuck** [jax-ml/jax#25212]
### Description

This error shows up recently in both our internal and external CI. We couldn't locate the culprit PR due to the mixture of recent JAX build failures.

A simple reproducer would be something like:
```
bazel test --//jax:build_jaxlib=false //tests/pallas:export_back_compat_pallas_test_gpu
```

Here is the error log from the latest JAX Toolbox CI: https://github.com/NVIDIA/JAX-Toolbox/actions/runs/12116482468/job/33777934258#step:7:952
```
[ RUN      ] CompatTest.test_triton_add_one
I1202 10:44:12.009619 139670423086912 export_back_compat_test_util.py:203] Lowering and running the function at the current version
I1202 10:44:12.187324 139670423086912 export_back_compat_test_util.py:208] Result of current version run is (array([1., 2., 3., 4., 5., 6., 7., 8.], dtype=float32),)
I1202 10:44:12.212428 139670423086912 export_back_compat_test_util.py:244] Writing the updated testdata at /root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/testlogs/tests/pallas/export_back_compat_pallas_test_gpu/test.outputs/test_triton_add_one.py
I1202 10:44:12.341165 139670423086912 export_back_compat_test_util.py:257] Running the serialized module
loc("-":0:0): error: expected mlir::arith::DenormalModeAttr, but got: #arith.fastmath<none>
2024-12-02 10:44:12.371269: E external/xla/xla/status_macros.cc:55] INTERNAL: RET_CHECK failure (external/xla/xla/service/gpu/ir_emitter_unnested.cc:1388) triton_module Failed to parse Triton module: ML�R
MLIRgooglex-trunk
[  FAILED  ] CompatTest.test_triton_add_one
======================================================================
ERROR: test_triton_add_one (__main__.CompatTest)
CompatTest.test_triton_add_one
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/tests/pallas/export_back_compat_pallas_test.py", line 61, in test_triton_add_one
    self.run_one_test(func, data)
  File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/internal_test_util/export_back_compat_test_util.py", line 258, in run_one_test
    res_run_serialized = self.run_serialized(
  File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/internal_test_util/export_back_compat_test_util.py", line 347, in run_serialized
    return pjit.pjit(exported.call)(*data.inputs)
  File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/traceback_util.py", line 180, in reraise_with_filtered_traceback
    return fun(*args, **kwargs)
  File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/pjit.py", line 337, in cache_miss
    pgle_profiler) = _python_pjit_helper(fun, jit_info, *args, **kwargs)
  File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/pjit.py", line 195, in _python_pjit_helper
    out_flat, compiled, profiler = _pjit_call_impl_python(*args_flat, **p.params)
  File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/pjit.py", line 1676, in _pjit_call_impl_python
    ).compile()
  File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/interpreters/pxla.py", line 2360, in compile
    executable = UnloadedMeshExecutable.from_hlo(
  File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/interpreters/pxla.py", line 2858, in from_hlo
    xla_executable = _cached_compilation(
  File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/interpreters/pxla.py", line 2674, in _cached_compilation
    xla_executable = compiler.compile_or_get_cached(
  File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/compiler.py", line 484, in compile_or_get_cached
    return _compile_and_write_cache(
  File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/compiler.py", line 716, in _compile_and_write_cache
    executable = backend_compile(
  File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/profiler.py", line 333, in wrapper
    return func(*args, **kwargs)
  File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/compiler.py", line 308, in backend_compile
    raise e
  File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/compiler.py", line 302, in backend_compile
    return backend.compile(built_c, compile_options=options)
jaxlib.xla_extension.XlaRuntimeError

----------------------------------------------------------------------
Ran 3 tests in 4.030s
```

Internally, python3 will complain about an UnicodeDecodeError`utf8` error
```
0: [ RUN      ] CompatTest.test_triton_add_one
0: I1202 12:04:15.499548 140737325779840 export_back_compat_test_util.py:203] Lowering and running the function at the current version
0: E1202 20:04:15.811356 3804888 ptx_compiler_helpers.cc:71] *** WARNING *** Invoking ptxas with version 12.3.107, which corresponds to a CUDA version <=12.6.2. CUDA versions 12.x.y up to and including 12.6.2 miscompile certain edge cases around clamping.
0: Please upgrade to CUDA 12.6.3 or newer.
0: I1202 12:04:15.982658 140737325779840 export_back_compat_test_util.py:208] Result of current version run is (array([1., 2., 3., 4., 5., 6., 7., 8.], dtype=float32),)
0: I1202 12:04:15.998625 14073732[577](https://gitlab-master.nvidia.com/dl/jet/ci/-/jobs/125776678#L577)9840 export_back_compat_test_util.py:244] Writing the updated testdata at /root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/testlogs/tests/pallas/export_back_compat_pallas_test_gpu/test.outputs/test_triton_add_one.py
0: I1202 12:04:17.923584 140737325779840 export_back_compat_test_util.py:257] Running the serialized module
0: loc("-":0:0): error: expected mlir::arith::DenormalModeAttr, but got: #arith.fastmath<none>
MLIRgooglex-trunk
0: [  FAILED  ] CompatTest.test_triton_add_one
0: ======================================================================
0: ERROR: test_triton_add_one (__main__.CompatTest)
0: CompatTest.test_triton_add_one
0: ----------------------------------------------------------------------
0: Traceback (most recent call last):
0:   File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/tests/pallas/export_back_compat_pallas_test.py", line 61, in test_triton_add_one
0:     self.run_one_test(func, data)
0:   File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/internal_test_util/export_back_compat_test_util.py", line 258, in run_one_test
0:     res_run_serialized = self.run_serialized(
0:                          ^^^^^^^^^^^^^^^^^^^^
0:   File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/internal_test_util/export_back_compat_test_util.py", line 347, in run_serialized
0:     return pjit.pjit(exported.call)(*data.inputs)
0:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
0:   File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/traceback_util.py", line 180, in reraise_with_filtered_traceback
0:     return fun(*args, **kwargs)
0:            ^^^^^^^^^^^^^^^^^^^^
0:   File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/pjit.py", line 337, in cache_miss
0:     pgle_profiler) = _python_pjit_helper(fun, jit_info, *args, **kwargs)
0:                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
0:   File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db[600](https://gitlab-master.nvidia.com/dl/jet/ci/-/jobs/125776678#L600)4de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/pjit.py", line 195, in _python_pjit_helper
0:     out_flat, compiled, profiler = _pjit_call_impl_python(*args_flat, **p.params)
0:                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
0:   File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/pjit.py", line 1676, in _pjit_call_impl_python
0:     ).compile()
0:       ^^^^^^^^^
0:   File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/interpreters/pxla.py", line 2360, in compile
0:     executable = UnloadedMeshExecutable.from_hlo(
0:                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
0:   File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/interpreters/pxla.py", line 2858, in from_hlo
0:     xla_executable = _cached_compilation(
0:                      ^^^^^^^^^^^^^^^^^^^^
0:   File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/interpreters/pxla.py", line 2674, in _cached_compilation
0:     xla_executable = compiler.compile_or_get_cached(
0:                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
0:   File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/compiler.py", line 484, in compile_or_get_cached
0:     return _compile_and_write_cache(
0:            ^^^^^^^^^^^^^^^^^^^^^^^^^
0:   File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/compiler.py", line 716, in _compile_and_write_cache
0:     executable = backend_compile(
0:                  ^^^^^^^^^^^^^^^^
0:   File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/profiler.py", line 333, in wrapper
0:     return func(*args, **kwargs)
0:            ^^^^^^^^^^^^^^^^^^^^^
0:   File "/root/.cache/bazel/_bazel_root/cfd1b2cc6fe180f3eb424db6004de364/execroot/__main__/bazel-out/k8-opt/bin/tests/pallas/export_back_compat_pallas_test_gpu.runfiles/__main__/jax/_src/compiler.py", line 302, in backend_compile
0:     return backend.compile(built_c, compile_options=options)
0:            ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
0: UnicodeDecodeError: 'utf-8' codec can't decode byte 0xef in position 134: invalid continuation byte
0: 
0: ----------------------------------------------------------------------
```

### System info (python version, jaxlib version, accelerator, etc.)

Basically the latest JAX and any type of GPUs.

```
jax:    0.4.36.dev20241202+385328b55
jaxlib: 0.4.36.dev20241202
numpy:  1.26.4
python: 3.11.10 (main, Sep  7 2024, 18:35:41) [GCC 11.4.0]
device info: NVIDIA A100-SXM4-80GB-8, 8 local devices"
process_count: 1
platform: uname_result(system='Linux', node='493553963e6b', release='5.15.0-1065-nvidia', version='#66-Ubuntu SMP Fri Aug 30 13:56:20 UTC 2024', machine='x86_64')
```
